engine/engine.go

```diff
@@ -47,36 +49,56 @@ type Config struct {
 	Workdir    string
 	ConfigPath string
 	// If true, do not load project extensions
-	NoExtensions  bool
-	LogOutput     io.Writer
-	JournalURI    string
-	JournalWriter journal.Writer
-	DisableHostRW bool
-	RunnerHost    string
-	SessionToken  string
-	UserAgent     string
-
-	// WARNING: this is currently exposed directly but will be removed or
-	// replaced with something incompatible in the future.
-	RawBuildkitStatus chan *bkclient.SolveStatus
+	NoExtensions   bool
+	LogOutput      io.Writer
+	JournalURI     string
+	JournalWriter  journal.Writer
+	ProgrockWriter progrock.Writer
+	DisableHostRW  bool
+	RunnerHost     string
+	SessionToken   string
+	UserAgent      string
 }
 
-type StartCallback func(context.Context, *router.Router) error
+type StartCallback func(context.Context, *progrock.Recorder, *router.Router) error
 
-func Start(ctx context.Context, startOpts *Config, fn StartCallback) error {
-	if startOpts == nil || startOpts.RunnerHost == "" {
+func Start(ctx context.Context, startOpts Config, fn StartCallback) error {
+	if startOpts.RunnerHost == "" {
 		return fmt.Errorf("must specify runner host")
 	}
 
+	if startOpts.ProgrockWriter == nil {
+		startOpts.ProgrockWriter = progrock.Discard{}
+	}
+
 	remote, err := url.Parse(startOpts.RunnerHost)
 	if err != nil {
 		return err
 	}
+
 	c, err := engine.NewClient(ctx, remote, startOpts.UserAgent)
 	if err != nil {
 		return err
 	}
 
+	if c.EngineName != "" && startOpts.LogOutput != nil {
+		fmt.Fprintln(startOpts.LogOutput, "Connected to engine", c.EngineName)
```

Comment:
@sipsma Moved this from `internal/engine/client.go` so it can use `LogOutput`; we need to be careful about printing to `os.Stderr` now since it messes with the inline TUI. I remember this was tied to a certain use case though, so not sure if this change is OK.

In the future we could add a Progrock vertex for internal logs (and maybe mark it "internal"), but this tweak is a smaller lift.

Reply:
Yeah I think this is good. The current use case was just to do a quick unblock of the desire to know which engine an SDK is connected to when looking at plain progress output directly (e.g. in GHA logs).

As long as it's still viewable in that context this is all good. What we should actually be doing here is emitting an event so that it shows up in the cloud UX (and presumably all the local TUIs too then?)